Cargo.toml

```diff
@@ -482,25 +482,27 @@ suspicious_double_ref_op = { level = "allow", priority = 2 }
 
 [workspace.lints.clippy]
 all = { level = "allow", priority = 0 }
-correctness = { level = "deny", priority = 1 }
+correctness = { level = "warn", priority = 1 }
+complexity = { level = "warn", priority = 1 }
 if-same-then-else = { level = "allow", priority = 2 }
-complexity = { level = "deny", priority = 1 }
 zero-prefixed-literal = { level = "allow", priority = 2 }            # 00_1000_000
 type_complexity = { level = "allow", priority = 2 }                  # raison d'etre
 nonminimal-bool = { level = "allow", priority = 2 }                  # maybe
 borrowed-box = { level = "allow", priority = 2 }                     # Reasonable to fix this one
 too-many-arguments = { level = "allow", priority = 2 }               # (Turning this on would lead to)
+needless-lifetimes = { level = "allow", priority = 2 }               # generated code
 unnecessary_cast = { level = "allow", priority = 2 }                 # Types may change
 identity-op = { level = "allow", priority = 2 }                      # One case where we do 0 +
 useless_conversion = { level = "allow", priority = 2 }               # Types may change
-unit_arg = { level = "allow", priority = 2 }                         # styalistic.
-option-map-unit-fn = { level = "allow", priority = 2 }               # styalistic
-bind_instead_of_map = { level = "allow", priority = 2 }              # styalistic
+unit_arg = { level = "allow", priority = 2 }                         # stylistic
+option-map-unit-fn = { level = "allow", priority = 2 }               # stylistic
+bind_instead_of_map = { level = "allow", priority = 2 }              # stylistic
 erasing_op = { level = "allow", priority = 2 }                       # E.g. 0 * DOLLARS
 eq_op = { level = "allow", priority = 2 }                            # In tests we test equality.
 while_immutable_condition = { level = "allow", priority = 2 }        # false positives
 needless_option_as_deref = { level = "allow", priority = 2 }         # false positives
 derivable_impls = { level = "allow", priority = 2 }                  # false positives
+clone_on_copy = { level = "allow", priority = 2 }                    # false positives
```

Comment:
I am just thinking, isn't there a way to allow these exceptions just for particular feature, e.g. `--features=runtime-benchmarks`? With something magic like `[target.'cfg(feature = "runtime-benchmarks")']`?

It would be maybe cool to have default clippy per workspace + additional exceptions per feature for better control, but also this is maybe over-engineering and unwanted.

Reply:
That would be really nice, but from what I understand we can't do that with this way of specifying clippy lints, that has to be applied in the `cargo/config.toml`, what do you think about having lints specified in two different places?

I tried adding exceptions per workspace, but it's not allowed - I think that the workspace lints can be thought of as an enforcement of a global style which cannot be overruled by the packages within the workspace, except line by line.

Reply:
@bkontur what are the false positives?

Reply:
> @bkontur what are the false positives?

I dont know for `clone_on_copy`, @seadanda found some cases for benchmarks, because actual clippy job does not check `--features runtime-benchmarks`.

But if we remove `clone_on_copy` and temporary remove `RUSTFLAGS: "-D warnings"` from clippy job, we should see all occurrences in CI

Reply:
The first false positives I noticed were in staking. I'm sure some of the existing `clone_on_copy` lints are legit and could be fixed, but I suspect it comes from clippy not dealing well with macros, especially multiple levels of macros inside macros and so will enforce workarounds for no real gain.